bumpversion/__init__.py

```diff
@@ -336,7 +342,62 @@ def __iter__(self):
         return iter(self._values)
 
     def __repr__(self):
-        return '<bumpversion.Version:{}>'.format(keyvaluestring(self._values))
+        by_part = ", ".join("{}={}".format(k, v) for k, v in self.items())
+        return '<bumpversion.Version:{}>'.format(by_part)
+
+    def __hash__(self):
+        return hash(tuple((k, v) for k, v in self.items()))
+
+    def _compare(self, other, method, strict=True):
+        """
+        For strict relations, if a part is equal we skip it
+        :param other: the other Version
+        :param method: the compare method
+        :return:
+        """
+        try:
+            for vals in ((v, other[k]) for k, v in self.items()):
+                if vals[0] == vals[1]:
+                    continue
+                if method(vals[0], vals[1]):
+                    return True
+                else:
+                    return False
+            return not strict
+        except KeyError:
+            raise TypeError("Versions use different parts, cant compare them.")
+
+    def __eq__(self, other):
+        if self is other:
+            return True
+        if not isinstance(other, Version):
+            return False
+        try:
+            return all(v == other[k] for k, v in self.items())
```

Comment:
Doesn't this ignore the case there `len(self.items()) < len(other.items())`?

Reply:
On a second though, I think equality should be strict, that is:  `1.4 != 1.4.0`, by assuming they are the same we ignore the version pattern defined by *parse* and *serialize*. If both versions where generated by bumpversion then this highlights that the user has an *error* in the part configuration.